### PR TITLE
include ec2 details in glooctl table output

### PR DESCRIPTION
**Pull request by mitchdraft**
Adds EC2 upstream details to table output
Allows creation of EC2 upstream through glooctl

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/988
resolves https://github.com/solo-io/gloo/issues/954

### CODE REVIEW

File: projects/gloo/cli/pkg/cmd/options/options.go [hunk 1]
```diff
@@ -68,6 +68,9 @@ type Upgrade struct {
 
 type Get struct {
 	Selector InputMapStringString
+	// if true, will get extended information that might not be available without calling external services, such as
+	// AWS apis, or kubectl port-forwarded services
+	Wide bool
```
Comment (mitchdraft) [inline projects/gloo/cli/pkg/cmd/options/options.go]: let me know how you feel about this new convention @ilackarms @EItanya

File: projects/gloo/cli/pkg/cmd/create/upstream.go [hunk 2]
```diff
@@ -180,6 +188,48 @@ func upstreamSpecFromOpts(input options.InputUpstream) (*v1.UpstreamSpec, error)
...
+		if svcSpec != nil {
+			return nil, errors.Errorf("%v does not support service spec", input.UpstreamType)
+		}
+		ec2Spec := &ec2.UpstreamSpec{
+			Region:   input.AwsEc2.Region,
+			Port:     input.AwsEc2.Port,
+			PublicIp: input.AwsEc2.PublicIp,
+			RoleArn:  input.AwsEc2.Role,
+		}
+		if input.AwsEc2.Secret.Namespace != "" || input.AwsEc2.Secret.Name != "" {
```
Comment (EItanya) [inline projects/gloo/cli/pkg/cmd/create/upstream.go]: is it valid not to have an AWS secret?

File: projects/gloo/cli/pkg/printers/upstream.go [hunk 3]
```diff
@@ -6,26 +6,32 @@ import (
 	"os"
 	"sort"
 
+	"github.com/solo-io/gloo/projects/gloo/cli/pkg/printers/types"
```
Comment (EItanya) [inline projects/gloo/cli/pkg/printers/upstream.go]: Extra spaces?

File: projects/gloo/cli/pkg/xdsinspection/ec2.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
+package xdsinspection
+
+import (
+	"github.com/solo-io/gloo/projects/gloo/pkg/plugins/aws/ec2"
+
+	"github.com/solo-io/gloo/projects/gloo/pkg/translator"
+	"github.com/solo-io/solo-kit/pkg/api/v1/resources/core"
+)
+
+func (xd *XdsDump) GetInstancesForUpstream(upstream core.ResourceRef) []string {
```
Comment (EItanya) [inline projects/gloo/cli/pkg/xdsinspection/ec2.go]: `GetEc2InstancesForUpstream`?

File: projects/gloo/cli/pkg/cmd/create/upstream.go (hunk 2 above)
Comment (mitchdraft) [reply projects/gloo/cli/pkg/cmd/create/upstream.go]: yep, will use the environment credentials if none provided, added a comment

**Pull request merged**